### PR TITLE
Add support for `outputs` section to uploader

### LINKED ISSUE #68: upload_or_check_non_existence uploads only 1 file
`conda-forge/openmotif-feedstock` ([meta.yaml](https://github.com/conda-forge/openmotif-feedstock/blob/master/recipe/meta.yaml)) is the first package to use the `outputs` section to generate multiple output packages. As expected, there are hick-ups. The first is that [upload_or_check_non_existence.py](https://github.com/conda-forge/conda-forge-build-setup-feedstock/blob/master/recipe/upload_or_check_non_existence.py) does not consider multiple output packages and only uploads the main `openmotif` tarball.

### PR DESCRIPTION
**Pull request by epruesse**
A recipe can be configured to produce multiple output packages via the
`outputs` section in `meta.yaml`. This patch makes
`upload_or_check_non_existence` honor this feature by looping over all
specified output packages when uploading to anaconda.

(fixes  conda-forge/conda-forge-build-setup-feedstock#68)

### CODE REVIEW

**Comment by jakirkham** [comment]
Needs a bump in the patch version (not build number).

**Comment by epruesse** [comment]
@jakirkham Done

**Comment by jakirkham** [comment]
Could you please fix the conflicts here, @epruesse?

**Comment by epruesse** [comment]
@jakirkham I think @isuruf was faster than me. I can rebase though if you want.

File: recipe/upload_or_check_non_existence.py [hunk 1]
```diff
@@ -103,22 +103,24 @@ def main():
...
-    if token:
-        if not exists:
-            upload(cli, meta, owner, channel)
-            print('Uploaded {}'.format(bldpkg_path(meta)))
+    meta_main = MetaData(recipe_dir)
+    for _, meta in meta_main.get_output_metadata_set(files=None):
+        print("Processing {}".format(meta.name()))
+        if meta.skip():
+            print("No upload to take place - this configuration was skipped in build/skip.")
+            return
```
Comment (isuruf) [inline recipe/upload_or_check_non_existence.py]: Shouldn't this be `continue` ?

Comment (epruesse) [reply recipe/upload_or_check_non_existence.py]: Good catch! Yes, it should.

**Comment by isuruf** [comment]
Have you tested this? If it works +1 to merge

**Comment by epruesse** [comment]
I've used it for [openmotif-feedstock](../../openmotif-feedstock) -- it worked fine there.

**Comment by jakirkham** [comment]
> I think @isuruf was faster than me. I can rebase though if you want.

He's pretty fast. 😄 No that's fine.

**Comment by jakirkham** [comment]
@conda-forge-linter, please lint.

**Comment by isuruf** [comment]
Thanks. I'm sorry it took this long. I can't even remember why I didn't merge.

**Pull request merged**